src/NinjaTools.FluentMockServer.API/Models/ViewModels/HttpRequestViewModel.cs

```diff
@@ -0,0 +1,67 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Microsoft.AspNetCore.Http;
+using Newtonsoft.Json;
+
+namespace NinjaTools.FluentMockServer.API.Models.ViewModels
+{
+    [JsonObject(ItemNullValueHandling = NullValueHandling.Ignore, ItemRequired = Required.AllowNull)]
+    public class HttpRequestViewModel
+    {
+        [JsonIgnore] public HttpRequest Request { get; }
+
+        public HttpRequestViewModel(HttpRequest request)
+        {
+            Request = request;
+            Body = ReadBody();
+
+            Headers = Request.Headers.Any()
+                ? Request.Headers.ToDictionary(
+                    k => k.Key,
+                    v => v.Value.ToList())
+                : null;
+
+            Cookies = request.Cookies.Any()
+                ? Request.Cookies.ToDictionary(
+                    k => k.Key,
+                    v => v.Value)
+                : null;
+
+            Method = Request.Method;
+            Path = Request.Path.Value;
+            QueryString = Request.QueryString.HasValue ? Request.QueryString.Value : null;
+            IsHttps = Request.IsHttps;
+        }
+
+        public string? Method { get; }
+        public string? Path { get; }
+        public string? Body { get; }
+        public string? QueryString { get; }
+        public Dictionary<string, List<string>> Headers { get; }
+        public Dictionary<string, string> Cookies { get; }
+        public bool IsHttps { get; }
+
+        private string? ReadBody()
+        {
+            var stream = Request.Body;
+
+            try
+            {
+                if (!stream.CanRead || !stream.CanSeek || stream.Length <= 0)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/git_36/NinjaTools.FluentMockServer/pullRequest?prid=4802644)